test/invariants.jl

```diff
@@ -1,7 +1,8 @@
 isdefined(@__MODULE__, :Memory) || const Memory = Vector # Compat for Julia < 1.11
 _get_UInt128(m::Memory, i::Integer) = UInt128(m[i]) | (UInt128(m[i+1]) << 64)
 _length_from_memory(allocated_memory::Integer) = Int((allocated_memory-10523)/7)
-function verify_weights(m::Memory)
+function verify_weights(w::DynamicDiscreteSamplers.Weights)
+    m = w.m
```

Comment:
This change seems unrelated and I'd rather not make it because it will make throwing in `verify(m)` calls into `src` internals harder which will reduce its utility in debugging